activerecord/lib/active_record/model_schema.rb

```diff
@@ -235,6 +235,29 @@ def table_name=(value)
         @predicate_builder = nil
       end
 
+      # Disabling automatic attribute discovery makes the model use only those attributes
+      # that have been explicitly defined with `ActiveRecord::Base.attribute` method.
+      # Defaults to `true`.
+      # Example:
+      #
+      #   class Project < ActiveRecord::Base
+      #     self.attribute_auto_discovery = false
+      #
+      #     attribute :name, :string
+      #     attribute :members_count, :integer
+      #   end
+      #
+      # Despite on the fact that `projects` table has more columns that `name` and `members_count`,
+      # ActiveRecord will only use explicitly declared attributes
+      def attribute_auto_discovery
```

Comment:
Can't be this moved to `included` block and defined via `mattr_accessor` or `class_attribute` with default value as another options?